xdsl/ir/core.py

```diff
@@ -563,7 +563,7 @@ class Operation(IRNode):
     _operands: tuple[SSAValue, ...] = field(default=())
     """The operation operands."""
 
-    results: list[OpResult] = field(default_factory=list)
+    results: tuple[OpResult, ...] = field(default=())
     """The results created by the operation."""
 
     successors: list[Block] = field(default_factory=list)
```

Comment:
seems like this should also be a tuple, but I wasn't sure whether this PR was the place to do that